### DISCUSSION TITLE
How do we add a language extension?

### DISCUSSION THREAD

**Discussion by vtnate** [zed-industries/zed#21021] [Feature Requests]
I've looked through the [zed docs](https://zed.dev/docs/extensions/languages) and there is a lot there that goes over my head. All I really want is Modelica syntax highlighting. I don't have Rust experience.

I'm shocked that in my transition from VSCode the only thing I really miss so far is this [modelica syntax highlighting](https://github.com/SimplyDanny/modelica-language-vscode) extension. Zed is pretty cool! That VSCode extension says it is based on [this SublimeText](https://github.com/BorisChumichev/modelicaSublimeTextPackage) package. I've also found OpenModelica has [something written up](https://github.com/OpenModelica/tree-sitter-modelica/) for tree-sitter, as well as [this tree-sitter grammar](https://github.com/modelicahub/tree-sitter-modelica) on ModelicaHub.

I'm happy to work on this, but I need some more guidance. Is there a canonical example repo I can refer to (how about [clojure](https://github.com/zed-industries/zed/tree/main/extensions/clojure))? If I were to make an extension, would I refer to any of the above links from the extension?

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
